test/extended/authentication/oidc.go

```diff
@@ -0,0 +1,580 @@
+
+				apiServerArgs := observedConfig["apiServerArguments"].(map[string]interface{})
+
+				o.Expect(apiServerArgs["authentication-token-webhook-config-file"]).To(o.BeNil(), "authentication-token-webhook-config-file argument should not be specified when OIDC authentication is configured")
+				o.Expect(apiServerArgs["authentication-token-webhook-version"]).To(o.BeNil(), "authentication-token-webhook-version argument should not be specified when OIDC authentication is configured")
+				o.Expect(apiServerArgs["authConfig"]).To(o.BeNil(), "authConfig argument should not be specified when OIDC authentication is configured")
+
+				o.Expect(apiServerArgs["authentication-config"]).NotTo(o.BeNil(), "authentication-config argument should be specified when OIDC authentication is configured")
+				o.Expect(apiServerArgs["authentication-config"].([]interface{})[0].(string)).To(o.Equal("/etc/kubernetes/static-pod-resources/configmaps/auth-config/auth-config.json"))
+			})
+
+			g.It("[Skipped] should remove the OpenShift OAuth stack", func() {
+				g.Skip("functionality not yet implemented")
+				/*
+					o.Eventually(func(gomega o.Gomega) {
+						_, err := oc.AdminKubeClient().AppsV1().Deployments("openshift-authentication").Get(context.TODO(), "oauth-openshift", metav1.GetOptions{})
+						gomega.Expect(err).NotTo(o.BeNil(), "should not be able to get the integrated oauth stack")
+						gomega.Expect(apierrors.IsNotFound(err)).To(o.BeTrue(), "integrated oauth stack should not be present when OIDC authentication is configured")
+					}).WithTimeout(20 * time.Minute).WithPolling(30 * time.Second).Should(o.Succeed())
+				*/
+			})
+
+			g.It("should not accept tokens provided by the OAuth server", func() {
+				o.Eventually(func(gomega o.Gomega) {
+					clientset, err := kubernetes.NewForConfig(oauthUserConfig)
+					gomega.Expect(err).NotTo(o.HaveOccurred())
+
+					_, err = clientset.CoreV1().Pods(oc.Namespace()).List(context.TODO(), metav1.ListOptions{})
+					gomega.Expect(err).ShouldNot(o.BeNil(), "should not be able to list Pods using OAuth client token")
+					gomega.Expect(apierrors.IsUnauthorized(err)).To(o.BeTrue(), "should receive an unauthorized error when trying to list Pods using OAuth client token")
+				}).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).Should(o.Succeed())
+			})
+
+			g.It("should accept tokens issued by the external IdP", func() {
+				// should always be able to create an SSAR for yourself
+				o.Eventually(func(gomega o.Gomega) {
+					// always re-authenticate to get a new token
+					err := keycloakCli.Authenticate("admin-cli", username, password)
+					gomega.Expect(err).NotTo(o.HaveOccurred(), "should not encounter an error authenticating as keycloak user")
+
+					copiedOC := *oc
+					tokenOC := copiedOC.WithToken(keycloakCli.AccessToken())
+
+					_, err = tokenOC.KubeClient().AuthorizationV1().SelfSubjectAccessReviews().Create(context.TODO(), &authzv1.SelfSubjectAccessReview{
```

Comment:
nit: Why not kill two birds with one stone with the next SSR test? Proves both token acceptance and cluster identity mapping.

Also, why not also use an SSR to prove the previous oauth token test as well?

Reply:
> nit: Why not kill two birds with one stone with the next SSR test? Proves both token acceptance and cluster identity mapping. We could still split it into two tests that validate these two things. 

Fair question. We can certainly perform a shared SSR.

> Also, why not also use an SSR to prove the previous oauth token test as well?

Certainly can. I think I just went with the GET on Pods because it was a quick and easy way to say "we are now unauthorized to get something we were previously authorized to get".

I think an SSR is much more reliable, so I'll switch it to use that.